## Using SQL Express Developer Edition on local laptop

**Stage 1, import original selections**

In [ ]:
IF OBJECT_ID ('TEMPDB..#ORIGINAL_SELECTION') IS NOT NULL DROP TABLE #ORIGINAL_SELECTION
GO
CREATE TABLE #ORIGINAL_SELECTION
(	
	CUSTOMER_ID int,
	LINK VARCHAR(250)
);

BULK INSERT #ORIGINAL_SELECTION
FROM 'C:\Users\napl\Downloads\customer_ids_links.csv'
WITH (
	DATAFILETYPE = 'char',
	FIRSTROW = 2,
	FIELDTERMINATOR = ',',
	ROWTERMINATOR = '\n' );

SELECT TOP 10 * FROM #ORIGINAL_SELECTION

**We need the value from the LINK column where the key is RID and store it in the same table so we can compare.**

In [ ]:
ALTER TABLE #ORIGINAL_SELECTION ADD RID BIGINT;
GO
UPDATE #ORIGINAL_SELECTION SET RID = REPLACE(LINK, 'https://bb.survey.com/G1122NL/survey/my_survey?name=G1122NL&RID=','');
GO
SELECT TOP 10 * FROM #ORIGINAL_SELECTION

**Import winners file**

In [ ]:
IF OBJECT_ID ('TEMPDB..#WINNERS') IS NOT NULL DROP TABLE #WINNERS
CREATE TABLE #WINNERS (
	[ID]	BIGINT
);
GO
BULK INSERT #WINNERS
FROM 'C:\Users\napl\Downloads\winners.csv'
WITH (
	DATAFILETYPE = 'char',
	FIRSTROW = 2,
	FIELDTERMINATOR = ',',
	ROWTERMINATOR = '\n' );
GO
SELECT TOP 10 * FROM #WINNERS;

**Check winners ID in relation to the original selection, this selection reveals the winners from the selection**

**Combine into an new temporary table, assign an identity column that we can use later..**

In [8]:
IF OBJECT_ID ('TEMPDB..#EXPORT_WINNERS') IS NOT NULL DROP TABLE #EXPORT_WINNERS
GO
CREATE TABLE #EXPORT_WINNERS (
    ID          INT IDENTITY(1,1) NOT NULL PRIMARY KEY,
    CUSTOMER_ID INT,
    CARDNUMBER  BIGINT NULL,
    PIN VARCHAR(4)
);
GO
INSERT INTO #EXPORT_WINNERS (CUSTOMER_ID)
SELECT  S.CUSTOMER_ID
FROM    #WINNERS W
JOIN    #ORIGINAL_SELECTION S ON W.ID = S.RID;
GO
SELECT TOP 10 * FROM #EXPORT_WINNERS

Commands completed successfully.

Commands completed successfully.

(150 rows affected)

(10 rows affected)

Total execution time: 00:00:00.036

ID CUSTOMER_ID CARDNUMBER PIN 1 4214941 NULL NULL 2 4276878 NULL NULL 3 4170876 NULL NULL 4 4207486 NULL NULL 5 4194813 NULL NULL 6 4033599 NULL NULL 7 3938711 NULL NULL 8 4252354 NULL NULL 9 4017893 NULL NULL 10 4121731 NULL NULL

**Import giftcard file**

In [ ]:
IF OBJECT_ID ('TEMPDB..#GIFTCARDS') IS NOT NULL DROP TABLE #GIFTCARDS
CREATE TABLE #GIFTCARDS
(	
	
	CARDNUMBER  BIGINT,
	PIN         VARCHAR(20),
	VALUE       INT
);
GO
BULK INSERT #GIFTCARDS
FROM 'C:\Users\napl\Downloads\giftcards.csv'
WITH (
    DATAFILETYPE = 'char',
    FIRSTROW = 2,
    FIELDTERMINATOR = ',',
    ROWTERMINATOR = '\n' );
GO
SELECT TOP 10 * FROM #GIFTCARDS

**Add identity column to the giftcards table, from 1 tom 150.... with that we have an relational key with the export\_winners table.**

In [9]:
ALTER TABLE #GIFTCARDS ADD ROWID INT IDENTITY(1,1) NOT NULL PRIMARY KEY;
GO

SELECT TOP 10 * FROM #GIFTCARDS

Commands completed successfully.

(10 rows affected)

Total execution time: 00:00:00.049

CARDNUMBER PIN VALUE ROWID 174943770439 9777 5 1 197827564076 6563 5 2 209293460449 4580 5 3 188081367736 6594 5 4 191237122026 2238 5 5 125154336924 4602 5 6 168101825465 5861 5 7 237409550065 7893 5 8 179823004372 2868 5 9 157866566156 7702 5 10

**Now we can join the winners and the giftcards to form 1 table, we update the corresponding values**

**Then every customer has been assigned an unique cardnumber and pincode**

In [10]:
SELECT *
FROM    #EXPORT_WINNERS E
JOIN    #GIFTCARDS G ON E.ID = G.ROWID

GO

(150 rows affected)

Total execution time: 00:00:00.021

ID CUSTOMER_ID CARDNUMBER PIN CARDNUMBER PIN VALUE ROWID 1 4214941 NULL NULL 174943770439 9777 5 1 2 4276878 NULL NULL 197827564076 6563 5 2 3 4170876 NULL NULL 209293460449 4580 5 3 4 4207486 NULL NULL 188081367736 6594 5 4 5 4194813 NULL NULL 191237122026 2238 5 5 6 4033599 NULL NULL 125154336924 4602 5 6 7 3938711 NULL NULL 168101825465 5861 5 7 8 4252354 NULL NULL 237409550065 7893 5 8 9 4017893 NULL NULL 179823004372 2868 5 9 10 4121731 NULL NULL 157866566156 7702 5 10 11 4220255 NULL NULL 145881301554 1449 5 11 12 4214007 NULL NULL 169585042248 2194 5 12 13 4184229 NULL NULL 106365139071 4294 5 13 14 4101518 NULL NULL 164477933000 4220 5 14 15 3697744 NULL NULL 247699045039 5655 5 15 16 4283002 NULL NULL 226314154428 4971 5 16 17 4216371 NULL NULL 193622256341 2569 5 17 18 4260321 NULL NULL 229730303234 2481 5 18 19 4217563 NULL NULL 201182376068 7497 5 19 20 4260233 NULL NULL 240264574863 9301 5 20 21 2425145 NULL NULL 247991954774 6821 5 21 22 4222947 NULL NULL 164612419452 6810 5 22 23 4019758 NULL NULL 147478570756 2491 5 23 24 4177668 NULL NULL 146533589239 1049 5 24 25 4249421 NULL NULL 209079603658 2124 5 25 26 4011025 NULL NULL 113159572622 9625 5 26 27 4243428 NULL NULL 190044250116 6473 5 27 28 4253550 NULL NULL 203300689253 4982 5 28 29 4028061 NULL NULL 115275670399 9573 5 29 30 4184756 NULL NULL 103240375611 2793 5 30 31 4270590 NULL NULL 220252293925 4119 5 31 32 4215111 NULL NULL 192507257273 5462 5 32 33 4224708 NULL NULL 171599420528 9341 5 33 34 3014806 NULL NULL 109856036155 7595 5 34 35 4254417 NULL NULL 227497716767 3006 5 35 36 4283255 NULL NULL 158222508015 9526 5 36 37 4254781 NULL NULL 186238036583 4047 5 37 38 4286785 NULL NULL 247724219317 6017 5 38 39 3454310 NULL NULL 247312211117 9589 5 39 40 4272644 NULL NULL 136231581620 9472 5 40 41 4196944 NULL NULL 107234638754 7687 5 41 42 4169990 NULL NULL 216975937247 1402 5 42 43 4209869 NULL NULL 194878242935 5410 5 43 44 4242201 NULL NULL 192057713162 9209 5 44 45 4220519 NULL NULL 204554724610 9109 5 45 46 4216374 NULL NULL 196058990701 2787 5 46 47 4247588 NULL NULL 163421689468 9653 5 47 48 4202536 NULL NULL 140934980053 7498 5 48 49 4239553 NULL NULL 120170500190 6535 5 49 50 4174193 NULL NULL 125277511160 8911 5 50 51 4248882 NULL NULL 196382363207 2297 5 51 52 4024896 NULL NULL 214600463425 8046 5 52 53 4209424 NULL NULL 180678656591 1166 5 53 54 4283508 NULL NULL 204173475200 9196 5 54 55 3919313 NULL NULL 115152048220 2965 5 55 56 4260356 NULL NULL 103292539828 3597 5 56 57 4214200 NULL NULL 235545858431 7084 5 57 58 4262346 NULL NULL 217024229514 5912 5 58 59 4059170 NULL NULL 158649956799 7513 5 59 60 4051148 NULL NULL 152574068414 8589 5 60 61 4198578 NULL NULL 196296312281 4370 5 61 62 4070770 NULL NULL 133731201791 7922 5 62 63 3970123 NULL NULL 204910242344 6149 5 63 64 4208679 NULL NULL 107052591844 4461 5 64 65 4265362 NULL NULL 198823573643 5773 5 65 66 4178970 NULL NULL 244533525832 6994 5 66 67 4250052 NULL NULL 224789977379 3703 5 67 68 4269267 NULL NULL 205263726139 6120 5 68 69 4260650 NULL NULL 216433098431 8526 5 69 70 3853128 NULL NULL 228550083877 6488 5 70 71 4186348 NULL NULL 193902813316 9798 5 71 72 4204398 NULL NULL 152987868945 7618 5 72 73 4233756 NULL NULL 176081272123 7412 5 73 74 4215705 NULL NULL 246371255218 8149 5 74 75 4193304 NULL NULL 227106996714 5522 5 75 76 4167726 NULL NULL 122002979292 3932 5 76 77 4254115 NULL NULL 201180436356 5757 5 77 78 4188282 NULL NULL 201905626904 3358 5 78 79 4240655 NULL NULL 162154926610 9165 5 79 80 3926398 NULL NULL 114637455300 3082 5 80 81 4186715 NULL NULL 180074881058 3972 5 81 82 4237953 NULL NULL 217195588937 1369 5 82 83 4175941 NULL NULL 155902594827 3530 5 83 84 4252507 NULL NULL 166139534358 5240 5 84 85 4185912 NULL NULL 111983916303 6209 5 85 86 4242908 NULL NULL 189626180682 6424 5 86 87 4235482 NULL NULL 134878717964 3762 5 87 88 3968537 NULL NULL 245053456277 2586 5 88 89 4203159 NULL NULL 166873555032 8392 5 89 90 3977610 NULL NULL 231443270852 1340 5 90 9

In [12]:
-- MAKE IT FINAL
UPDATE E 
    SET  CARDNUMBER = G.CARDNUMBER,
        PIN = G.PIN
FROM    #EXPORT_WINNERS E
JOIN    #GIFTCARDS G ON E.ID = G.ROWID
GO

(150 rows affected)

Total execution time: 00:00:00.004

**Get the final table**

In [14]:
SELECT CUSTOMER_ID, CARDNUMBER, PIN
FROM #EXPORT_WINNERS

(150 rows affected)

Total execution time: 00:00:00.016

CUSTOMER_ID CARDNUMBER PIN 4214941 174943770439 9777 4276878 197827564076 6563 4170876 209293460449 4580 4207486 188081367736 6594 4194813 191237122026 2238 4033599 125154336924 4602 3938711 168101825465 5861 4252354 237409550065 7893 4017893 179823004372 2868 4121731 157866566156 7702 4220255 145881301554 1449 4214007 169585042248 2194 4184229 106365139071 4294 4101518 164477933000 4220 3697744 247699045039 5655 4283002 226314154428 4971 4216371 193622256341 2569 4260321 229730303234 2481 4217563 201182376068 7497 4260233 240264574863 9301 2425145 247991954774 6821 4222947 164612419452 6810 4019758 147478570756 2491 4177668 146533589239 1049 4249421 209079603658 2124 4011025 113159572622 9625 4243428 190044250116 6473 4253550 203300689253 4982 4028061 115275670399 9573 4184756 103240375611 2793 4270590 220252293925 4119 4215111 192507257273 5462 4224708 171599420528 9341 3014806 109856036155 7595 4254417 227497716767 3006 4283255 158222508015 9526 4254781 186238036583 4047 4286785 247724219317 6017 3454310 247312211117 9589 4272644 136231581620 9472 4196944 107234638754 7687 4169990 216975937247 1402 4209869 194878242935 5410 4242201 192057713162 9209 4220519 204554724610 9109 4216374 196058990701 2787 4247588 163421689468 9653 4202536 140934980053 7498 4239553 120170500190 6535 4174193 125277511160 8911 4248882 196382363207 2297 4024896 214600463425 8046 4209424 180678656591 1166 4283508 204173475200 9196 3919313 115152048220 2965 4260356 103292539828 3597 4214200 235545858431 7084 4262346 217024229514 5912 4059170 158649956799 7513 4051148 152574068414 8589 4198578 196296312281 4370 4070770 133731201791 7922 3970123 204910242344 6149 4208679 107052591844 4461 4265362 198823573643 5773 4178970 244533525832 6994 4250052 224789977379 3703 4269267 205263726139 6120 4260650 216433098431 8526 3853128 228550083877 6488 4186348 193902813316 9798 4204398 152987868945 7618 4233756 176081272123 7412 4215705 246371255218 8149 4193304 227106996714 5522 4167726 122002979292 3932 4254115 201180436356 5757 4188282 201905626904 3358 4240655 162154926610 9165 3926398 114637455300 3082 4186715 180074881058 3972 4237953 217195588937 1369 4175941 155902594827 3530 4252507 166139534358 5240 4185912 111983916303 6209 4242908 189626180682 6424 4235482 134878717964 3762 3968537 245053456277 2586 4203159 166873555032 8392 3977610 231443270852 1340 3637546 214024965081 4277 4196461 157866566156 7702 3780352 175234849008 5398 3945168 249331929857 1092 4207921 186026764980 4100 3857269 118808370130 6050 4232788 123024767543 9261 3595761 238320092716 3856 4245422 163187648704 7115 3919389 237594696951 5223 3917225 216284471218 8235 4173229 193746177194 4327 4103338 174732721445 8519 4223595 108074649224 8240 4270830 183297638118 7719 3605332 220448039938 6310 4051125 199942140747 9361 4211103 119582914442 7162 4231735 154142241794 3747 4161908 113265673968 7133 4036322 117538452673 3693 4170758 118229569665 4926 4208638 177037995499 6496 4189637 196608221417 1999 4203916 158761932044 8813 4249639 144011463749 2976 4060544 227353075568 3478 4246089 233691969014 6324 4287972 102941604341 9780 4200269 234754175222 9245 4275760 180342564128 1409 4169366 140974810788 8860 4264863 113017050089 7263 4284271 107780115565 3852 4196347 206514101468 9889 3543979 170576499184 6155 4188190 207963067908 2731 4247721 167226161871 8806 4201832 198555648825 6458 4196837 166020512099 1617 4178049 142535697152 1499 4188592 196614338740 6128 4198009 168036533092 3551 4280589 227407725089 2685 4242045 134574519691 9705 2769559 140139730274 3263 4251447 183558086837 2879 4097111 205383640024 9401 4203875 148283413484 1078 2719030 240715120191 6533 4232205 120380643658 3844 4256946 236696525571 4128 4244497 119935081930 5736 4218564 114353751619 2584 3962454 106991701856 9338 4266651 155144413108 4499 3501800 116470852500 8449 4258673 228088106191 2429 4133137 125444447593 6191 4209254 234726042481 6839